samza-core/src/main/java/org/apache/samza/storage/TaskRestoreManager.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.storage;
+
+import java.util.Map;
+import org.apache.samza.system.SystemStreamPartition;
+
+
+/**
+ * The helper interface restores task state.
+ */
+public interface TaskRestoreManager {
+
+  /**
+   * Init state resources such as file directories.
+   */
+  void init(Map<SystemStreamPartition, String> checkpointedChangelogSSPOffsets);
+
+  /**
+   * Restore state from checkpoints, state snapshots and changelog.
+   */
+  void restore();
+
+  /**
+   * Stop all persistent stores after restoring.
+   */
+  void stopPersistentStores();
```

Comment:
Just retaining existing implementation and behavior. Since this is an internal interface (in samza-core), we can update this in future (e.g. during a ContainerStorageManager cleanup).